subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +391,52 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+    @Unroll
+    def "report setters for property with mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                // no field, just getter and setter
+                @InputFiles public ${type} getMutablePropertyWithSetter() { return null; } 
+                public void setMutablePropertyWithSetter(${type} value) {} 
+
+                // no field, just getters
+                @InputFiles public ${type} getMutablePropertyWithoutSetter() { return null; } 
+
+                // no field, just setter
+                public void setMutablePropertyWithoutGetter() {} 
+
+                // field with getter and setter
+                private ${type} fieldWithGetterAndSetter;                
+                @InputFiles public ${type} getFieldWithGetterAndSetter() { return null; } 
+                public void setFieldWithGetterAndSetter(${type} value) {} 
+                
+                // final field with constructor and getter
+                private final ${type} finalFieldWithoutSetter; 
+                MyTask() { finalFieldWithoutSetter = null; } 
+                @InputFiles public ${type} getSources() { return null; } 
```

Comment:
Fields don't come into play here at all, so we should not mention them in this test. Let's remove them (`fieldWithGetterAndSetter` and `finalFieldWithoutSetter`).

Reply:
Done